README.md

```diff
@@ -1,21 +1,124 @@
-# LiveX
+# LiveEx - State Management for Phoenix LiveViews
 
-**TODO: Add description**
+LiveEx is a State Management library for [Phoenix LiveView](https://github.com/phoenixframework/phoenix_live_view).
+
+LiveEx is based on the [Flux pattern](https://github.com/facebook/flux/tree/master/examples/flux-concepts)
+and its implementation is inspired by the [Vuex](https://vuex.vuejs.org/) library.
+
+LiveEx helps organize the state changes and data flow in LiveView frontends.
+State changes (aka. `Actions`) are forced through a `Dispatcher` (or funnel) in a sequencial FIFO manner before
+the changes are applied to a global state managed by a `Store`. The same store can be shared by multiple LiveViews.
```

Comment:
I am missing more details on the problem you are trying to solve here (shared state between multiple live views?)